python/nvfuser/benchmark_utils.py

```diff
@@ -22,67 +46,109 @@ def cleanup(self):
         pass
 
 
-class TorchProfileTimer(Timer):
+def cupti_call_safe(func, *args):
+    """Wrapper for CUPTI calls. Failing CUPTI calls will exit the program."""
+    try:
+        return func(*args)
+    except Exception as e:
+        print(f"CUPTI call {func.__name__} failed: {e}")
+        sys.exit(1)
+
+
+class CuptiProfiler:
+    # List of activities to be recorded by CUPTI.
+    activity_kinds = [
+        cupti.ActivityKind.CONCURRENT_KERNEL,
+    ]
+    _subscriber_handle = None
+
+    @staticmethod
+    def enable_cupti_activities():
```

Comment:
should this have been a class method instead? since we are accessing CuptiProfiler.xxx

Reply:
more importantly, should we have some safety mutex-like thing here so that:
1. only one instance would enter the active profiling state
2. only the active instance would be able to tear down the active state